src/distilabel/steps/task/ultracm.py

```diff
@@ -0,0 +1,140 @@
+---
+
+### Instruction
+{{ instruction }}
+
+### Answer
+{{ completion }}
+---
+
+Please act as a teacher and provide specific and constructive feedback. Besides describing the weaknesses of the answer, you should also provide specific suggestions to guide me toward understanding how to improve. Please note, however, that your suggestions should help me better complete the instructions, but you should not introduce new requirements that are not mentioned in the instructions. Your feedback should focus on enhancing my ability to think critically and respond accurately. However, never explicitly provide the reference answer, nor do polite phrases be required. Only respond with concise feedback in chat style. Finally, score the overall quality of the answer from 1 to 10, where 1 is the worst and 10 is the best.
+
+*Format*
+### Feedback
+[Your feedback]
+Overall Score: [1-10]
+—--
+
+### Feedback
+"""
+
+
+class UltraCM(Task):
+    """A critique specialized `Task` following the prompt templated used by UltraCM (from UltraFeedback).
+
+    By default will be initialized with a specific `LLM` model from Hugging Face Hub using `vLLM`, which
+    corresponds to the `openbmb/UltraCM-13b` model, the model that was trained to be used with this task.
+    Take into account that independent of the engine used for the model, it's prepared to work with this specific one.
+
+    Input columns:
+        - `instruction` (`str`): The instruction that was used to generate the `completion`.
+        - `completion` (`str`): The instruction that was used to generate the `completion`.
+
+    Output columns:
+        - `score` (`float`): The overall score of the answer from 1 to 10.
+        - `critique` (`str`): The feedback given from the model to improve the answer.
+        - `raw_output` (`str`): The raw output from the model, in case it couldn't be parsed.
+
+    Notes:
+        Since the UltraCM model has been trained with OpenAI API generated data, the prompting
+        strategy may just be consistent / compliant with either GPT-3.5 or GPT-4 from OpenAI API, or
+        with their own model. Any other model may fail on the generation of a structured output, as
+        well as providing an incorrect / inaccurate critique.
+
+    References:
+        - [`UltraFeedback: Boosting Language Models with High-quality Feedback`](https://arxiv.org/abs/2310.01377)
+        - [`UltraFeedback - GitHub Repository`](https://github.com/OpenBMB/UltraFeedback)
+        - [`openbmb/UltraCM-13b`](https://huggingface.co/openbmb/UltraCM-13b)
+    """
+
+    llm: LLM = vLLM(model="openbmb/UltraCM-13b")
+    system_prompt: str = (
+        "User: A one-turn chat between a curious user and an artificial intelligence"
+        " assistant. The assistant gives helpful, very detailed, and polite answers to"
+        " the user's questions.</s>"
```

Comment:
They don't have a template in the Hub, I tried removing it and the generated text is off, it inserts extra tokens, so even if it's weird it seems to be necessary (from the model card: https://huggingface.co/openbmb/UltraCM-13b)

Reply:
It's their EOS token, but it feels weird that's injected within the `system_prompt` already, as that should ideally be handled by the `chat_template` from the `tokenizer`, see https://huggingface.co/openbmb/UltraCM-13b/blob/main/special_tokens_map.json#L9

Reply:
Also since they are not using a `chat_template` I'm assuming it's just a plain text generation, but maybe we can replace that with `tokenizer.eos_token` or something similar? As some LLMs may use a different `eos_token`

Reply:
Since this is a specific model that should not be used with a different LLM/tokenizer that the one it was trained, and accessing the tokenizer may be different depending on the LLM engine we use (`vLLM` has the tokenizer behind `get_tokenizer` method and `Transformers` has it under tokenizer for example), I think it's more direct just assigning the value after the prompt than getting the `eos_token` depending on the engine.